### ISSUE TITLE
Linter: set literals

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#1411]
Meta-issue corresponding to https://github.com/dart-lang/language/issues/37, the implementation of set literals.

- [x] update `prefer_collection_literals` (#1413)
- [x] update `prefer_const_declarations` to be aware of implicit const Set literals (#1427)

**Comment by pq** [comment]
A nuance:

```dart
var s = new Set(); 
```

doesn't imply the simple literal:

```dart
var s = {}; 
```

(since that's actually a map 🤔.)

If a set literal is preferred in this case, I guess we have two options:

1. `Set s = {};`, or
2. `var s = <dynamic>{};`

@munificent: thinking ahead to readability, what would you suggest?

**Comment by munificent** [comment]
The latter is how [Effective Dart suggests you write the code](https://www.dartlang.org/guides/language/effective-dart/design#avoid-redundant-type-arguments-on-generic-invocations).

**Comment by pq** [comment]
Thanks!

**Comment by dballesteros7** [comment]
Is it intended that the linter already triggers on code like:

```
final x = Set<String>();
```

Even though Set literals are not yet released? See this CI build:
https://travis-ci.org/dnfield/flutter_svg/builds/496594380

**Comment by pq** [comment]
Hi @dballesteros7.  A linter with `prefer_collection_literals` updated to flag Set literals rolled into the SDK right after the feature was enabled by default.  Since Flutter pulls SDK versions irrespective of official releases that might be the source of the confusion.  (The feature should be enabled in that SDK though.)
